allennlp/nn/util.py

```diff
@@ -567,3 +567,42 @@ def logsumexp(tensor: torch.Tensor,
     else:
         stable_vec = tensor - max_score.unsqueeze(dim)
     return max_score + (stable_vec.exp().sum(dim, keepdim=keepdim)).log()
+
+
+def add_bos_eos(tensor, mask, bos, eos):
```

Comment:
Add types here and in the docstring.

Reply:
Maybe mention ELMo in the docstring here too.

Reply:
ok